src/components/input/input.js

```diff
@@ -795,6 +798,8 @@ function ngMessagesDirective() {
     if (attrs.mdAutoHide == 'false' || hasVisibiltyDirective(attrs)) {
       element.toggleClass('md-auto-hide', false);
     }
+
+    inputContainer.element.toggleClass('md-input-has-messages', true);
```

Comment:
This is the same as `inputContainer.element.addClass('md-input-has-messages');`.